### ISSUE TITLE
Unhandled Promise rejection: TextHighlighter is already enabled.

### ISSUE THREAD

**Issue by skyleguy** [stephanrauh/ngx-extended-pdf-viewer#1501]
- Does the bug also happen on https://mozilla.github.io/pdf.js/web/viewer.html? (If so, please open the issue at Mozilla's bugtracker).

No

- Does the bug also happen on https://pdfviewer.net/simple?

No

- Can you write a reproducer based on this template project? https://github.com/stephanrauh/ngx-extended-pdf-viewer-issues/tree/main/template

Unfortunately the PDFs we are using contain confidential info so I cannot supply the PDF I am using but can supply the options im passing to the extended pdf viewer

**Describe the bug**
When opening the viewer with documents from my company I am seeing this error happen frequently (but not always)

**Version info**
- Version of ngx-extended-pdf-viewer: 15.0.0-alpha.4 and 13.5.2

**Desktop (please complete the following information):**
 - Chrome

**To Reproduce**

1. use version 15.0.0-alpha.4 (also present in 13.5.2)
2. setup pdf viewer like so:
```
    <ngx-extended-pdf-viewer
      [src]="pdfSrc"
      [showToolbar]="false"
      [handTool]="false"
      [textLayer]="true"
      [showBorders]="false"
      zoom="page-width"
      backgroundColor="rgb(242 242 242)"
      [page]="1"
    ></ngx-extended-pdf-viewer>
```
3. PDF displays fine but error displays in console when pdf is rendered

**Screenshots**

![Screen Shot 2022-08-19 at 1 10 44 PM](https://user-images.githubusercontent.com/14967057/185671784-aa7ede9c-0d69-4be6-9dc5-af8c270a37f9.png)

**Comment by stephanrauh** [comment]
This bug (or nuisance - as far as I can see, it's just a warning) doesn't show on my machine, so I need your help. Please add the attribute `[minifiedJSLibraries]="false"` so you can debug. Add a breakpoint in the viewer*.js at this line:
https://github.com/stephanrauh/pdf.js/blob/b834a0ca61f28800f3c7b83b0bc522dc4230383d/web/text_highlighter.js#L65

I suppose it's interesting to observe the calling function. The text highlighter is enabled twice, but I don't understand how that's possible. I suspect there's a text highlighter for each page, and maybe it's activated both for the text layer and for XFA forms. Does you PDF file use such a form?

**Comment by skyleguy** [comment]
the particular pdf im using only has one page actually.

After setting the breakpoint there i do see it get hit twice. The first time the value of textDivs and textContentItemsStr are both empty arrays and the enabled field is false. The second time the textDivs and textContentItemsStr are both filled with 119 element/strings and then the error gets thrown because enabled is already true

**Comment by stephanrauh** [comment]
I have to admit I'm not familiar with that part of the implementation of pdf.js. I'm just the guy providing an Angular wrapper for it. So I can't decide whether the empty arrays are OK or not. My gut says it sounds fishy, but that's just a guess.

What about the stack trace - can you spot a difference there?

Plus, I wonder if the bug has something to do with your PDF document. Here's a small selection of demo PDFs: https://github.com/stephanrauh/extended-pdf-viewer-showcase/tree/main/src/assets/pdfs. Does the bug happen with these PDF files, too?

**Comment by skyleguy** [comment]
interestingly enough it does not seem to happen on a few of the pdfs with text that ive tried from that folder... really not sure whats going on there. Even more strange it does not happen to my document every time either. im not seeing anything else to indicate the issue in the stacktrace unfortunately

**Comment by skyleguy** [comment]
i seem to have discovered that when the app is actually built and deploy this issue no longer happens with documents it previously had issues with on local dev. Very bizarre

**Comment by skyleguy** [comment]
im extremely unfamiliar with the expected format of PDFs, do you have any ideas as to what I could look for in our internal PDFs that may be causing this issue to happen?

**Comment by stephanrauh** [comment]
I'm almost as unfamiliar with the format of PDF files as you are! :)

In any case, you can remove the error message yourself: https://pdfviewer.net/extended-pdf-viewer/filtering-console-log

**Comment by stephanrauh** [comment]
I've removed the error message. Now the code doesn't throw an exception. I don't think that's going to cause problems, but I'd appreciate your test (once I've published version 15.0.0-alpha.6).

**Comment by skyleguy** [comment]
sure ill check it out once that version drops. interestingly enough im unable to see that error being logged from the log filter (which im currently using to filter out those warn messages i mentioned in another issue I brought up)

**Comment by stephanrauh** [comment]
The error message is thrown as an exception, so maybe it slips the attention of my filter implementation.

Yesterday, Tim caught me by surprise when he released a new version of pdf.js, so I guess the final release of version 15 is delayed another few days.

**Comment by stephanrauh** [comment]
I've just published version 15.0.0-alpha.6. Two-way binding of forms is broken (#1516), but I guess that's not your use-case, so please run a test.
